pkg/engine/image.go

```diff
@@ -0,0 +1,87 @@
+package engine
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/containers/podman/v4/pkg/bindings/images"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"k8s.io/klog/v2"
+)
+
+const imageMethod = "image"
+
+// Image configures targets to run a system prune periodically
+type Image struct {
+	CommonMethod `mapstructure:",squash"`
+	// Url is the url of the image to be loaded onto the system
+	Url string `mapstructure:"url"`
+}
+
+func (i *Image) GetKind() string {
+	return imageMethod
+}
+
+func (i *Image) SchedInfo() SchedInfo {
+	return SchedInfo{
+		schedule: i.Schedule,
+		skew:     i.Skew,
+	}
+}
+
+func (i *Image) Process(ctx, conn context.Context, PAT string, skew int) {
+	target := i.GetTarget()
+	time.Sleep(time.Duration(skew) * time.Millisecond)
+	target.mu.Lock()
+	defer target.mu.Unlock()
+
+	err := i.loadPodman(ctx, conn, i.Url)
+	if err != nil {
+		klog.Warningf("Repo: %s Method: %s encountered error: %v, resetting...", target.Name, imageMethod, err)
+	}
+
+}
+
+func (i *Image) MethodEngine(ctx context.Context, conn context.Context, change *object.Change, path string) error {
+	return i.loadPodman(ctx, conn, i.Url)
+}
+
+func (i *Image) Apply(ctx, conn context.Context, currentState, desiredState plumbing.Hash, tags *[]string) error {
+	changeMap, err := applyChanges(ctx, i.GetTarget(), i.GetTargetPath(), currentState, desiredState, tags)
+	if err != nil {
+		return err
+	}
+	if err := runChangesConcurrent(ctx, conn, i, changeMap); err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
can return nil for these, like the Clean method does (like @josephsawaya mentioned): https://github.com/containers/fetchit/blob/main/pkg/engine/clean.go#L42-#L67